golem-worker-executor-base/src/services/invocation_key.rs

```diff
@@ -1,274 +1,264 @@
-                .expect("failed to send confirmation");
-        }
-    }
-
-    fn resume_key(&self, worker_id: &WorkerId, key: &InvocationKey) {
-        self.cleanup();
-        let key = (worker_id.clone(), key.clone());
-        let mut state = self.state.lock().unwrap();
-        if let Some(status) = state.pending_keys.get_mut(&key) {
-            status.interrupted = false;
-        }
-    }
-
-    async fn wait_for_confirmation(
-        &self,
-        worker_id: &WorkerId,
-        key: &InvocationKey,
-    ) -> LookupResult {
-        debug!("wait_for_confirmation for {worker_id}: {key:?}");
-        loop {
-            match self.lookup_key(worker_id, key) {
-                LookupResult::Invalid => break LookupResult::Invalid,
-                LookupResult::Interrupted => break LookupResult::Interrupted,
-                LookupResult::Pending => {
-                    let expected_key: Option<(WorkerId, InvocationKey)> =
-                        Some((worker_id.clone(), key.clone()));
-                    let mut receiver = self.confirm_sender.subscribe();
-                    let confirmed_key = receiver.recv().await.ok();
-                    if confirmed_key == expected_key {
-                        break self.lookup_key(worker_id, key);
-                    } else {
-                        continue;
-                    }
-                }
-                LookupResult::Complete(result) => break LookupResult::Complete(result),
-            }
-        }
-    }
-}
-
-#[cfg(test)]
-mod tests {
-    use golem_common::model::{ComponentId, WorkerId};
-    use golem_wasm_rpc::Value;
-
-    use crate::services::invocation_key::{
-        InvocationKeyService, InvocationKeyServiceDefault, LookupResult,
-    };
-
-    #[cfg(test)]
-    #[test]
-    fn replay_in_same_order_works() {
-        let svc1 = InvocationKeyServiceDefault::default();
-        let uuid = uuid::Uuid::parse_str("14e55083-2ff5-44ec-a414-595a748b19a0").unwrap();
-
-        let worker_id = WorkerId {
-            component_id: ComponentId(uuid),
-            worker_name: "1".to_string(),
-        };
-
-        let key1 = svc1.generate_key(&worker_id);
-        let key2 = svc1.generate_key(&worker_id);
-        let key3 = svc1.generate_key(&worker_id);
-
-        let svc2 = InvocationKeyServiceDefault::default();
-        svc2.confirm_key(&worker_id, &key1, Ok(vec![Value::U32(1)]));
-        svc2.confirm_key(&worker_id, &key2, Ok(vec![Value::U32(2)]));
-        svc2.confirm_key(&worker_id, &key3, Ok(vec![Value::U32(3)]));
-
-        let r1 = svc2.lookup_key(&worker_id, &key1);
-        let r2 = svc2.lookup_key(&worker_id, &key2);
-        let r3 = svc2.lookup_key(&worker_id, &key3);
-
-        assert_eq!(r1, LookupResult::Complete(Ok(vec![Value::U32(1)])));
-        assert_eq!(r2, LookupResult::Complete(Ok(vec![Value::U32(2)])));
-        assert_eq!(r3, LookupResult::Complete(Ok(vec![Value::U32(3)])));
-    }
-}
+// // Copyright 2024 Golem Cloud
```

Comment:
should not we remove this file if everything is commented?

Reply:
we should, thanks